### PR TITLE
add and fix handle_service tests

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

**Comment by Tianhao-Gu** [comment]
@MrCreosote Read for review. Thanks!

File: .classpath [hunk 1]
```diff
@@ -53,8 +53,7 @@
 	<classpathentry kind="lib" path="/jars/lib/jars/mysql/mysql-connector-java-5.1.22-bin.jar"/>
```
Comment (MrCreosote) [inline .classpath]: Needs:
version bump to 0.9.1 (in the server class and docsource/conf.py)
release notes (create a new 0.9.1 version in the notes) (see ./docsource)
documentation updates since the deploy configuration is different (see ./docsource)
update to deploy.cfg.example
update to deploy template in deployment dir

Comment (MrCreosote) [inline .classpath]: Needs:
version bump to 0.9.1 (in the server class and docsource/conf.py)
release notes (create a new 0.9.1 version in the notes) (see ./docsource)
documentation updates since the deploy configuration is different (see ./docsource)
update to deploy.cfg.example
update to deploy template in deployment dir

File: src/us/kbase/common/service/ServiceChecker.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
 	public static final int PY_CODE = -32700;
 	public static final String JV_PL_MSG = "HTTP GET not allowed.";
 	public static final int JV_PL_CODE = -32300;
+	public static final String HandleSerivce_MSG = "No JSON object could be decoded";
```
Comment (MrCreosote) [inline src/us/kbase/common/service/ServiceChecker.java]: Was the handle service built with an older version of the SDK? I just had to change the JSON object line to the PY_MSG line above when the catalog was updated to python3: 
https://github.com/kbase/workspace_deluxe/commit/c04d5a809dcea0e942a6dab52f1de598759cf1ac

File: src/us/kbase/workspace/WorkspaceServer.java [hunk 3]
```diff
@@ -121,7 +121,7 @@
 	private final Types types;
 	private final WorkspaceAdministration wsadmin;
 	
-	private final URL handleManagerUrl;
+	private final URL handleServiceUrl;
```
Comment (MrCreosote) [inline src/us/kbase/workspace/WorkspaceServer.java]: I think the handle service url can be deleted as it's stored in WorkspaceServerMethods and the checking method is called there. AFAICT the check method below is now unused.

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java [hunk 4]
```diff
@@ -100,22 +99,22 @@ public boolean isFailed() {
 		private final WorkspaceAdministration wsadmin;
 		private final Types types;
 		private final BasicShockClient linkedShockClient;
-		private final URL handleManagerUrl;
+		private final URL handleServiceUrl;
```
Comment (MrCreosote) [inline src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: I don't think the handle service url is needed here either given my comments on the server class.

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java [hunk 5]
```diff
@@ -488,56 +484,31 @@ private static AuthToken getHandleToken(
...
-					handleManagerUrl, handleMgrToken);
-			if (handleManagerUrl.getProtocol().equals("http")) {
-				rep.reportInfo("Warning - the Handle Manager url uses insecure http. " +
-						"https is recommended.");
-				cli.setIsInsecureHttpConnectionAllowed(true);
-			}
-		} catch (Exception e) {
-			rep.reportFail("Could not establish a connection to the Handle Manager Service at "
-					+ handleManagerUrl + ": " + e.getMessage());
+			rep.reportFail("Could not establish a connection to the Handle Service at "
```
Comment (MrCreosote) [inline src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: This message is misleading - initializing the client doesn't validate anything. I think an IOException is specified for the constructor but if you look at the client code is never actually thrown. That's why I left the isOwner() call in the same block.

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java [hunk 6]
```diff
@@ -488,56 +484,31 @@ private static AuthToken getHandleToken(
...
+					+ handleServiceUrl + ": " + e.getMessage());
 			return null;
 		}
 		try {
-			cli.setPublicRead(Arrays.asList("FAKEHANDLE_-100"));
+			cli.isOwner(Arrays.asList("FAKEHANDLE_-100"));
```
Comment (MrCreosote) [inline src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: Nice, this is better.

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java [hunk 7]
```diff
@@ -488,56 +484,31 @@ private static AuthToken getHandleToken(
...
 		} catch (Exception e) {
-			if (!(e instanceof ServerException) || !e.getMessage().contains(
-							"Unable to set acl(s) on handles FAKEHANDLE_-100")) {
-				rep.reportFail("Could not establish a connection to the Handle Manager Service at "
-						+ handleManagerUrl + ": " + e.getMessage());
-				return null;
-			}
+
+			rep.reportFail("Could not successfullly run methods on the Handle Service at "
```
Comment (MrCreosote) [inline src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: successfully

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java [hunk 8]
```diff
@@ -488,56 +484,31 @@ private static AuthToken getHandleToken(
 		return null;
 	}
 	
-
-	private static void checkHandleServiceConnection(
+	private static AbstractHandleClient getHandleServiceClient(
```
Comment (MrCreosote) [inline src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: Have you tested the failure modes for this method manually, since there aren't any tests (and adding tests would be a massive pain?)

File: .classpath [hunk 9]
```diff
@@ -53,8 +53,7 @@
...
 	<classpathentry kind="lib" path="/jars/lib/jars/kbase/kidl/kbase-kidl-parser-1409261812-7863aef.jar"/>
 	<classpathentry kind="lib" path="/jars/lib/jars/jsemver/java-semver-0.9.0.jar"/>
-	<classpathentry kind="lib" path="/jars/lib/jars/kbase/handle/HandleManagerClient-160803-08c8da3.jar"/>
-	<classpathentry kind="lib" path="/jars/lib/jars/kbase/handle/HandleServiceClient-160803-b9de699.jar"/>
+	<classpathentry kind="lib" path="/jars/lib/jars/kbase/handle/AbstractHandleClient-1.0.0.jar"/>
```
Comment (MrCreosote) [inline .classpath]: Get rid of the SQL jar here and in build.xml

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java [hunk 10]
```diff
@@ -35,115 +34,118 @@
...
-	private final Process handleManager;
-	private final int handleManagerPort;
 	
 	private final Path tempDir;
 	
-	private final static String DB = "hsi";
-	private final static String TABLE = "Handle";
-	private final static String USER = "hsitest";
-	private final static String PWD = "hsi-pass-test";
+	private final static String DB = "handle_db";
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: I think this should be a constructor argument to avoid possible name collisions. Allows the user to specify the db name.

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java [hunk 11]
```diff
@@ -35,115 +34,118 @@
...
 	public HandleServiceController(
-			final String plackupExe,
-			final String abstractHandlePSGIpath,
-			final String handleManagerPSGIpath,
-			final String handleManagerAllowedUser,
-			final MySQLController mysql,
+			final MongoController mongo,
 			final String shockHost,
 			final AuthToken shockAdminToken,
 			final String perl5lib,
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: pretty sure you don't need the perl lib location

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java [hunk 12]
```diff
@@ -35,115 +34,118 @@
...
+				":" + handleServicePort, "--wsgi-file",
+				"AbstractHandle/AbstractHandleServer.py", "--pythonpath", lib_dir_path)
 				.redirectErrorStream(true)
 				.redirectOutput(tempDir.resolve("handle_service.log").toFile());
 		Map<String, String> env = handlepb.environment();
 		env.put("PERL5LIB", perl5lib);
-		env.put("KB_DEPLOYMENT_CONFIG", hsIniFile.toString());
+		env.put("KB_DEPLOYMENT_CONFIG", hsIniFile.getAbsolutePath().toString());
 		env.put("KB_SERVICE_NAME", HANDLE_SERVICE_NAME);
+		env.put("KB_AUTH_TOKEN", shockAdminToken.getToken());
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: Why is this in the environment?

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java [hunk 13]
```diff
@@ -35,115 +34,118 @@
...
 		
-		ProcessBuilder handlemgrpb = new ProcessBuilder(plackupExe, "--port",
-				"" + handleManagerPort, handleManagerPSGIpath)
-				.redirectErrorStream(true)
-				.redirectOutput(tempDir.resolve("handle_mngr.log").toFile());
-		env = handlemgrpb.environment();
-		env.put("PERL5LIB", perl5lib);
-		env.put("KB_DEPLOYMENT_CONFIG", hmIniFile.toString());
-		handleManager = handlemgrpb.start();
+		String handle_repo_prefix = "https://raw.githubusercontent.com/kbase/handle_service2/develop/lib/AbstractHandle/";
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: I think I would rather leave it up to the user to set up the handle service files so that the handle service version doesn't change during testing unexpectedly, and so that we can test against older versions if we're trying to debug.

I'd just make a helper script or instructions for setting up the handle service appropriately so the tests are runnable, and add a configuration item in test.cfg that points to the proper directory. We may want the user to be responsible for setting up the python path if that's necessary to get virtualenvs to work correctly, not sure about that.

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java [hunk 14]
```diff
@@ -154,14 +156,17 @@ private File createHandleServiceDeployCfg(
...
-		hs.add("mysql-host", "127.0.0.1");
-		hs.add("mysql-port", "" + mysql.getServerPort());
-		hs.add("mysql-user", USER);
-		hs.add("mysql-pass", PWD);
-		hs.add("data-source", "dbi:mysql:" + DB);
+		hs.add("mongo-host", "127.0.0.1");
+		hs.add("mongo-port", "" + mongo.getServerPort());
+		hs.add("mongo-database", DB);
+		hs.add("mongo-collection", COLLECTION);
+		hs.add("admin-roles", handleAdminRole);
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: Why do you need to specify the collection?

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java [hunk 15]
```diff
@@ -169,36 +169,28 @@ public static void setUpClass() throws Exception {
...
-		
 		HANDLE = new HandleServiceController(
-				WorkspaceTestCommon.getPlackupExe(),
-				WorkspaceTestCommon.getHandleServicePSGI(),
-				WorkspaceTestCommon.getHandleManagerPSGI(),
-				"user3",
-				MYSQL,
+				MONGO,
 				shockURL.toString(),
-				HANDLE_MNGR_TOKEN,
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: Why change the token here? I didn't want to mix up the test tokens with the handle service token.

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java [hunk 16]
```diff
@@ -333,37 +333,33 @@ public static void tearDownClass() throws Exception {
...
 	public void status() throws Exception {
 		final Map<String, Object> st = CLIENT1.status();
-		
+
 		//top level items
 		assertThat("incorrect state", st.get("state"), is((Object) "OK"));
 		assertThat("incorrect message", st.get("message"), is((Object) "OK"));
-		// should throw an error if not a valid semver
-		Version.valueOf((String) st.get("version")); 
+		// should throw an error if not a valid sever
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: semver was correct

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java [hunk 17]
```diff
@@ -391,15 +387,29 @@ private void checkMem(final Object num, final String name)
...
+		final ShockNode node = bsc.addNode();
+		String shock_id = node.getId().getId();
+		final Handle handle = new Handle();
+		handle.setId(shock_id);
+		handle.setFileName("empty_file");
+		handle.setUrl(bsc.getShockUrl().toString());
+		handle.setType("shock");
+		String handle_id = UUID.randomUUID().toString();
+		System.out.println("generated random handle_id for testing: " + handle_id);
+		handle.setHid(handle_id);
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: Does the handle service normally allow setting the ID? I thought the ID was an auto incrementing index key with the 'KBH_' prefix.

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java [hunk 18]
```diff
@@ -417,84 +427,62 @@ public void basicHandleTest() throws Exception {
...
+		bsc.addToNodeAcl(new ShockNodeId(shock_id), Arrays.asList(USER2), ShockACLType.ALL);
 		String workspace2 = "basichandle2";
 		CLIENT2.createWorkspace(new CreateWorkspaceParams().withWorkspace(workspace2));
 		try {
 			CLIENT2.saveObjects(new SaveObjectsParams().withWorkspace(workspace2)
 					.withObjects(Arrays.asList(
 							new ObjectSaveData().withData(new UObject(handleobj)).withName("foo2")
 							.withType(HANDLE_TYPE))));
-			fail("saved object with bad handle");
-		} catch (ServerException e) {
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: This should not have changed. This operation should fail and the assert below should succeed.

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java [hunk 19]
```diff
@@ -417,84 +427,62 @@ public void basicHandleTest() throws Exception {
...
 				.withUsers(Arrays.asList(USER2)).withNewPermission("r"));
 		//get objects2
-				
 		ObjectData ret1 = CLIENT2.getObjects2(new GetObjects2Params()
 				.withObjects(Arrays.asList(new ObjectSpecification()
 					.withWorkspace(workspace)
 					.withObjid(1L)))).getData().get(0);
 		checkExternalIDError(ret1.getHandleError(), ret1.getHandleStacktrace());
-		
-		checkReadAcl(node, twouser);
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: Why are these lines removed?

Comment (MrCreosote) [reply src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: And all the other similar lines below that are removed.

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java [hunk 20]
```diff
@@ -506,32 +494,23 @@ public void basicHandleTest() throws Exception {
...
 		ObjectData wod = CLIENT2.getObjects2(new GetObjects2Params()
 				.withObjects(Arrays.asList(new ObjectSpecification()
 					.withWorkspace(workspace)
 					.withObjid(1L)))).getData().get(0);
-		
+
 		@SuppressWarnings("unchecked")
 		Map<String, Object> retdata = wod.getData().asClassInstance(Map.class);
 		assertThat("got correct data", retdata, is(handleobj));
-
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: These errors are expected by the test. Why did this change? It seems like many tests have been removed.

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java [hunk 21]
```diff
@@ -759,66 +738,67 @@ public void publicWorkspaceHandleTest() throws Exception {
 		String workspace = "publicWS";
 		CLIENT1.createWorkspace(new CreateWorkspaceParams()
 				.withWorkspace(workspace).withGlobalread("r"));
-		Handle h1 = HANDLE_CLIENT.newHandle();
+
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: Same questions for this test. A number of assertions were removed, which implies things aren't acting as they're supposed to.

File: src/us/kbase/workspace/WorkspaceServer.java (hunk 3 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/WorkspaceServer.java]: 👍

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java (hunk 4 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: 👍

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java (hunk 5 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: 👍

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java (hunk 7 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: 👍

File: .classpath (hunk 9 above)
Comment (Tianhao-Gu) [reply .classpath]: 👍

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java (hunk 10 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: 👍

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java (hunk 11 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: 👍

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java (hunk 15 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: 👍

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java (hunk 16 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: 👍

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java (hunk 12 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: I want to pass token in config so that my helper class can have token. Otherwise, I guess I can pass ctx['token'] to each method call. 
https://github.com/kbase/handle_service2/blob/develop/lib/AbstractHandle/AbstractHandleImpl.py#L38

File: src/us/kbase/common/service/ServiceChecker.java (hunk 2 above)
Comment (Tianhao-Gu) [reply src/us/kbase/common/service/ServiceChecker.java]: hmmm handle service was build with latest SDK in python3

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java (hunk 14 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: I wanted to reserve a specific collection just for handles.
https://github.com/kbase/handle_service2/blob/develop/deploy.cfg#L18

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java (hunk 17 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: 👍 you are right. I don't need to set handle id.

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java (hunk 8 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: I didn't test it. yeaaa it would be very painful to test it.

File: src/us/kbase/common/service/ServiceChecker.java (hunk 2 above)
Comment (MrCreosote) [reply src/us/kbase/common/service/ServiceChecker.java]:
```
curl https://ci.kbase.us/services/catalog
{"error": {"code": -32700, "name": "Parse error", "message": "Expecting value: line 1 column 1 (char 0)", "error": null}, "version": "1.1"}
```

Comment (MrCreosote) [reply src/us/kbase/common/service/ServiceChecker.java]: Maybe check with James on this? I don't see why the catalog and HS should give different errors for the GET verb.

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java (hunk 8 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: You should at least test it manually. What I do is start the tests with a debugger and add a break point after the services are all up but the WS isn't. Then I find the pid of the handle services and kill them and let the tests continue.

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java (hunk 12 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: I don't understand how this is supposed to work... KB_AUTH_TOKEN will be set by the env when the service starts up, right? So if you use that token in service methods it won't correspond with the user that's actually calling the method. Maybe I don't understand what it's being used for?

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java (hunk 14 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: Why does it need to be configurable though? The HS owns the database, so it can just decide what collection it wants to use. DBs aren't (or at least shouldn't be) shared between applications.

File: src/us/kbase/common/service/ServiceChecker.java (hunk 2 above)
Comment (Tianhao-Gu) [reply src/us/kbase/common/service/ServiceChecker.java]: 👍 I got it fixed running Python3 locally and on travis.

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java (hunk 12 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: 👍 I got what you meant. removed `KB_AUTH_TOKEN` from environ.

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java (hunk 14 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: 👍 removed collection config.

File: .classpath (hunk 1 above)
Comment (Tianhao-Gu) [reply .classpath]: 👍

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java (hunk 13 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: 👍

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java (hunk 18 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: 👍

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java (hunk 19 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: 👍

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java (hunk 20 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: 👍

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java (hunk 21 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: 👍

**Comment by Tianhao-Gu** [comment]
@MrCreosote Could you give it another review? I didn't set up handle service correctly before.

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java (hunk 8 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: 👍

File: .classpath [hunk 22]
```diff
@@ -50,11 +50,9 @@
 	<classpathentry kind="lib" path="/jars/lib/jars/logback/logback-classic-1.1.2.jar"/>
 	<classpathentry kind="lib" path="/jars/lib/jars/apache_commons/velocity-1.7.jar"/>
 	<classpathentry kind="lib" path="/jars/lib/jars/apache_commons/commons-lang-2.4.jar"/>
```
Comment (MrCreosote) [inline .classpath]: Needs:
* update documentation describing with what handle service the WSS is compatible (https://ci.kbase.us/services/ws/docs/buildandconfigure.html#service-dependencies) and update the configuration docs (https://ci.kbase.us/services/ws/docs/buildandconfigure.html#handle-manager-url)
* version bump to 0.9.1 in the server class (https://github.com/kbase/workspace_deluxe/blob/develop/src/us/kbase/workspace/WorkspaceServer.java#L111)

File: .travis.yml [hunk 23]
```diff
@@ -3,18 +3,30 @@ sudo: required
...
+  - MONGODB_VER=mongodb-linux-x86_64-2.6.12 ANT_TEST=test_quick_coverage WIRED_TIGER=false
+  - MONGODB_VER=mongodb-linux-x86_64-3.6.10 ANT_TEST=test_quick_coverage WIRED_TIGER=false
+  - MONGODB_VER=mongodb-linux-x86_64-3.6.10 ANT_TEST=test_quick_coverage WIRED_TIGER=true
 
 before_install:
   - sudo apt-get -qq update
   - sudo apt-get install -y ant-optional
-  - sudo pip install sphinx
-  
+  - sudo apt-get install -y build-essential python3-pip python3-dev
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply .travis.yml]: 👍

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java [hunk 24]
```diff
@@ -100,22 +99,20 @@ public boolean isFailed() {
...
-		private final URL handleManagerUrl;
 		
 		public WorkspaceInitResults(
 				final Workspace ws,
 				final WorkspaceServerMethods wsmeth,
 				final WorkspaceAdministration wsadmin,
 				final Types types,
 				final BasicShockClient linkedShockClient,
-				final URL handleManagerUrl) {
+				final URL handleServiceUrl) {
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java]: 👍

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java [hunk 25]
```diff
@@ -32,223 +31,169 @@
...
-			final MySQLController mysql,
+			final MongoController mongo,
 			final String shockHost,
 			final AuthToken shockAdminToken,
-			final String perl5lib,
 			final Path rootTempDir,
-			URL authServiceURL)
+			final URL authURL,
+			final String handleAdminRole,
+			final Path handleServiceDir)
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: 👍

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java [hunk 26]
```diff
@@ -506,32 +506,29 @@ public void basicHandleTest() throws Exception {
...
 		@SuppressWarnings("unchecked")
 		Map<String, Object> retdata = wod.getData().asClassInstance(Map.class);
 		assertThat("got correct data", retdata, is(handleobj));
-		
-		assertThat("got correct error message", wod.getHandleError(),
-				is("The Handle Manager reported a problem while attempting to set Handle ACLs: Unable to set acl(s) on handles "
-						+ h1.getHid()));
+		assertThat("got correct error message", wod.getHandleError(), 
+						startsWith("The Handle Manager reported a problem while attempting to set Handle ACLs: 'Grant readable access for node failed."));
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: 👍 make error message same as old handle manager service.

File: src/us/kbase/workspace/test/kbase/HandleAndShockTest.java [hunk 27]
```diff
@@ -759,27 +756,38 @@ public void publicWorkspaceHandleTest() throws Exception {
...
+		String shock_id = node.getId().getId();
+
+		final Handle handle = new Handle();
+		handle.setId(shock_id);
+		handle.setFileName("empty_file");
+		handle.setUrl(bsc.getShockUrl().toString());
+		handle.setType("shock");
+		String handle_id = UUID.randomUUID().toString();
+		System.out.println("generated random handle_id for testing: " + handle_id);
+		handle.setHid(handle_id);
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/kbase/HandleAndShockTest.java]: 👍 yup

File: .classpath (hunk 22 above)
Comment (Tianhao-Gu) [reply .classpath]: 👍

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java [hunk 28]
```diff
@@ -32,223 +31,169 @@
...
+					biokbase_dir.resolve(file_name).toFile());
+		}
+
+		Path log_file = null;
+		Stream <String> lines = null;
+		List <String> replaced = null;
+
+		log_file = handle_dir.resolve("AbstractHandleServer.py");
+		lines = Files.lines(log_file);
+		replaced = lines.map(line -> line.replaceAll("loads\\(request_body\\)",
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: 👍 removed.

**Comment by Tianhao-Gu** [comment]
@MrCreosote Can you give it another review? Thanks!

File: .travis.yml [hunk 29]
```diff
@@ -3,23 +3,44 @@ sudo: required
...
   - cd ..
   - git clone https://github.com/kbase/jars
   - export JARSDIR=`pwd`/jars/lib/jars/
-  - cd -
+  - export HS_VER=version_1.0.0
+  - git clone -b $HS_VER --single-branch https://github.com/kbase/handle_service2.git
+  - cd handle_service2/lib/
+  - mkdir biokbase
+  - cd biokbase/
+  - wget https://raw.githubusercontent.com/kbase/sdkbase2/python/log.py
```
Comment (MrCreosote) [inline .travis.yml]: This should be versioned as well, even if it's just going against a commit

File: deploy.cfg.example [hunk 30]
```diff
@@ -27,11 +27,10 @@ auth-service-url = https://kbase.us/services/auth/api/legacy/KBase/Sessions/Logi
 # The KBase authorization service mkII url.
 auth2-service-url = https://kbase.us/services/auth
 
-# The urls for the Handle Service and Handle Manager.
+# The urls for the Handle Service.
 handle-service-url =
-handle-manager-url =
-# The token used to access the handle manager.
-handle-manager-token = 
+# The token used to access the handle manager (shock admin token).
```
Comment (MrCreosote) [inline deploy.cfg.example]: This is not necessarily a shock admin token - it's a token that allows the user to use the set_read_acl (or whatever, I don't remember the name exactly) method on the Handle Manager (now Handle Service). The Handle Service presumably has a token that *is* a shock admin, but this token may or may not be.

File: deployment/conf/.templates/deployment.cfg.templ [hunk 31]
```diff
@@ -21,15 +21,14 @@ temp-dir={{ default .Env.temp_dir "ws_temp_dir" }}
...
-handle-manager-url={{ default .Env.handle_manager "https://ci.kbase.us/services/handlemngr/" }}
 # The KBase authorization server url for the legacy KBase endpoint.
 auth-service-url={{ default .Env.auth_service "https://ci.kbase.us/services/auth/api/legacy/KBase" }}
 auth-service-url-allow-insecure={{ default .Env.auth_service_url_allow_insecure "" }}
 # The KBase authorization server url.
 auth2-service-url = {{ default .Env.auth2_service "https://ci.kbase.us/services/auth/" }}
 auth2-ws-admin-read-only-roles = {{ default .Env.auth2_ws_admin_read_only_roles "" }}
 auth2-ws-admin-full-roles = {{ default .Env.auth2_ws_admin_full_roles "" }}
-#the credentials used for the handle manager.
+#the credentials used for the handle manager (shock admin token).
```
Comment (MrCreosote) [inline deployment/conf/.templates/deployment.cfg.templ]: Same as above.

File: docsource/buildandconfigure.rst [hunk 32]
```diff
@@ -214,25 +211,19 @@ ignore-handle-service
...
-**Required**: If using handles
-
-**Description**: The URL of the Handle Manager
-
 handle-manager-token
 """"""""""""""""""""
 **Required**: If using handles
 
-**Description**: Credentials for the account approved for Handle Manager use
+**Description**: Credentials for the account approved for Shock admin use
```
Comment (MrCreosote) [inline docsource/buildandconfigure.rst]: Same as above

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java [hunk 33]
```diff
@@ -32,223 +31,160 @@
...
-		hs.add("data-source", "dbi:mysql:" + DB);
-		
+		hs.add("auth-url", authURL.toString());
+		hs.add("shock-url", shockHost);
+		hs.add("admin-token", shockAdminToken.getToken().toString());
+		hs.add("admin-roles", handleAdminRole);
+
+		hs.add("mongo-host", "127.0.0.1");
+		hs.add("mongo-port", "" + mongo.getServerPort());
+		hs.add("mongo-database", "handle_db");
```
Comment (MrCreosote) [inline src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: Can you make this a constructor param so it can be configured or make is something that's less likely to have a name collision? Maybe `handle_controller_test_handle_db` or something

File: .travis.yml (hunk 29 above)
Comment (Tianhao-Gu) [reply .travis.yml]: 👍

File: deploy.cfg.example (hunk 30 above)
Comment (Tianhao-Gu) [reply deploy.cfg.example]: 👍

File: deployment/conf/.templates/deployment.cfg.templ (hunk 31 above)
Comment (Tianhao-Gu) [reply deployment/conf/.templates/deployment.cfg.templ]: 👍

File: docsource/buildandconfigure.rst (hunk 32 above)
Comment (Tianhao-Gu) [reply docsource/buildandconfigure.rst]: 👍

File: src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java (hunk 33 above)
Comment (Tianhao-Gu) [reply src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java]: 👍

**Comment by Tianhao-Gu** [comment]
@MrCreosote 🌟 🌟 🌟 all done  🌟 🌟 🌟
